.github/workflows/ci.yml

```diff
@@ -0,0 +1,316 @@
+    runs-on: ubuntu-latest
+    timeout-minutes: 60
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Get submodules hash
+        id: submodules
+        run: |
+          echo "hash=$(git submodule status | awk '{print $1}' | sort | shasum -a 256 | sed 's/[ -]*//g')" >> $GITHUB_OUTPUT
+
+      - name: Cache submodules
+        uses: actions/cache@v3
+        with:
+          path: |
+            vendor/
+            .git/modules
+          key: ${{ runner.os }}-vendor-modules-${{ steps.submodules.outputs.hash }}
+
+      - name: Read VERSION file
+        id: waku_version
+        run: |
+          VERSION=$(cat ./VERSION)
+          echo "version=$VERSION" >> $GITHUB_OUTPUT
+
+      - name: Install Go
+        uses: actions/setup-go@v4
+        with:
+          go-version: ${{ env.GO_VERSION }} 
+          cache: false
+
+      - name: Build
+        run: |
+          make install-gomobile
+          make mobile-android || make mobile-android
+          cd ./build/lib
+          tar -czvf gowaku-${{ steps.waku_version.outputs.version }}-android.tar.gz gowaku.aar gowaku-sources.jar 
+
+      - name: Upload asset
+        uses: actions/upload-artifact@v3
+        with:
+          name: gowaku-${{ steps.waku_version.outputs.version }}-android.tar.gz
+          path: ./build/lib/gowaku-${{ steps.waku_version.outputs.version }}-android.tar.gz
+          if-no-files-found: error
+
+  build-ios:
+    needs: changes
+    if: ${{ needs.changes.outputs.v2 == 'true' || needs.changes.outputs.common == 'true' }}
+    runs-on: macos-latest
+    timeout-minutes: 60
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Get submodules hash
+        id: submodules
+        run: |
+          echo "hash=$(git submodule status | awk '{print $1}' | sort | shasum -a 256 | sed 's/[ -]*//g')" >> $GITHUB_OUTPUT
+
+      - name: Cache submodules
+        uses: actions/cache@v3
+        with:
+          path: |
+            vendor/
+            .git/modules
+          key: ${{ runner.os }}-vendor-modules-${{ steps.submodules.outputs.hash }}
+
+      - name: Read VERSION file
+        id: waku_version
+        run: |
+          VERSION=$(cat ./VERSION)
+          echo "version=$VERSION" >> $GITHUB_OUTPUT
+
+      - name: Install Go
+        uses: actions/setup-go@v4
+        with:
+          go-version: ${{ env.GO_VERSION }} 
+          cache: false
+
+      - name: Build
+        run: |
+          make install-gomobile
+          make mobile-ios
+          cd ./build/lib
+          tar -czvf gowaku-${{ steps.waku_version.outputs.version }}-ios.tar.gz Gowaku.xcframework
+
+      - name: Upload asset
+        uses: actions/upload-artifact@v3
+        with:
+          name: gowaku-${{ steps.waku_version.outputs.version }}-ios.tar.gz
+          path: ./build/lib/gowaku-${{ steps.waku_version.outputs.version }}-ios.tar.gz
+          if-no-files-found: error
+
+
+  test:
```

Comment:
Wondering if it makes sense to keep all jobs in 1 file called ci.yml or have separate files and refer them in main file.